Lib/fontTools/varLib/mutator.py

```diff
@@ -121,6 +122,46 @@ def instantiateVariableFont(varfont, location, inplace=False):
 		log.info("Building interpolated tables")
 		merger.instantiate()
 
+	addidef = False
+	for glyph in glyf.glyphs.values():
+		if hasattr(glyph, "program"):
+			instructions = glyph.program.getAssembly()
+			# If GETVARIATION opcode is used in bytecode of any glyph add IDEF
+			addidef = any(op.startswith("GETVARIATION") for op in instructions)
+			if addidef:
+				break
+
+	if addidef:
+		log.info("Adding IDEF to fpgm table for GETVARIATION opcode")
+		asm = []
+		if 'fpgm' in varfont:
+			fpgm = varfont['fpgm']
+			asm = fpgm.program.getAssembly()
+		else:
+			fpgm = newTable('fpgm')
+			fpgm.program = ttProgram.Program()
+			varfont['fpgm'] = fpgm
+		asm.append("PUSHB[000] 145")
+		asm.append("IDEF[ ]")
+		args = [str(len(loc))]
+		for a in fvar.axes:
+			args.append(str(floatToFixed(loc[a.axisTag], 14)))
+		asm.append("NPUSHW[ ] " + ' '.join(args))
+		asm.append("ENDF[ ]")
+		fpgm.program.fromAssembly(asm)
+
+		# Change maxp attributes as IDEF is added
+		if 'maxp' in varfont:
+			maxp = varfont['maxp']
+			if hasattr(maxp, "maxInstructionDefs"):
+				maxp.maxInstructionDefs += 1
+			else:
+				setattr(maxp, "maxInstructionDefs", 1)
```

Comment:
Don't know if we need the else blocks.  If maxp table is version 0.5 then we just don't touch it?

LGTM otherwise.

Reply:
Right. But quotation is what to do if it isn't :).  Current code is fine.

Reply:
I think we should unconditionally assume that maxInstructionDefs attribute is present in a TrueType maxp table, and let it raise AttributeError in the unlikely event that the attribute is missing, either because it’s a maxp version 0.5, or because the table hasn’t been fully constructed/decompiled.

Reply:
Ignoring what spec says, it's perfectly fine using a version .5 for TrueType fonts.  The rest of fields in 1.0 are unused by most moderns systems.  I wouldn't want something like mutator police what people use.

Reply:
alright.. in that case, setting the attribute seems unnecessary because it won't be compiled in the end.